src/main/java/reactor/core/publisher/WorkQueueProcessor.java

```diff
@@ -482,6 +510,36 @@
 				autoCancel);
 	}
 
+	/**
+	 * Create a new WorkQueueProcessor using the passed buffer size, wait strategy
+	 * and auto-cancel settings. <p> A Shared Processor authorizes concurrent onNext calls
+	 * and is suited for multi-threaded publisher that will fan-in data. <p> The passed
+	 * {@link ExecutorService} will execute as many event-loop
+	 * consuming the ringbuffer as subscribers.
+	 * <p> An additional {@code requestTaskExecutor} {@link ExecutorService} is also used
+	 * internally on each subscription.
+	 * @param executor A provided ExecutorService to manage threading infrastructure
+	 * @param requestTaskExecutor A provided ExecutorService to manage threading infrastructure
+	 * @param bufferSize A Backlog Size to mitigate slow subscribers
+	 * @param strategy A RingBuffer WaitStrategy to use instead of the default
+	 * smart blocking wait strategy.
+	 * @param autoCancel Should this propagate cancellation when unregistered by all
+	 * subscribers ?
+	 * @param <E> Type of processed signals
+	 * @return a fresh processor
+	 */
+	public static <E> WorkQueueProcessor<E> share(ExecutorService executor,
+			ExecutorService requestTaskExecutor,
```

Comment:
Who's responsibility is it to shutdown this executor?  The other executor is shutdown gracefully as part of EventLoopProcessor completion/shutdown, but this one isn't.  Two concerns there: 
i) inconsistency
ii) request thread won't be stopped gracefully as part of processor shutdown.

Side note: This is IMO one advantage to passing `Supplier<ExecutorService>` as it make the responsibility clearer. I probably wouldn't do that here though as it would be inconsistent unless you broke API.